src/pages/organize/[orgId]/campaigns/[campId]/calendar.tsx

```diff
@@ -42,27 +48,53 @@ export const getServerSideProps : GetServerSideProps = scaffold(async (ctx) => {
     }
 }, scaffoldOptions);
 
-type CampaignCalendarPageProps = {
+type OrganizeCalendarPageProps = {
     campId: string;
     orgId: string;
 };
 
-const CampaignCalendarPage: PageWithLayout<CampaignCalendarPageProps> = ({ orgId, campId }) => {
+const CampaignCalendarPage : PageWithLayout<OrganizeCalendarPageProps> = ({ orgId, campId }) => {
     const eventsQuery = useQuery(['campaignEvents', orgId, campId], getCampaignEvents(orgId, campId));
     const events = eventsQuery.data || [];
+    const intl = useIntl();
+
+    const items = [
+        {
+            name: intl.formatMessage(
+                { id: 'misc.calendar.month' }),
+        },
+        {
+            name: intl.formatMessage(
+                { id: 'misc.calendar.week' }),
+        },
+    ];
+    const [calendarView, setCalendarView] = useState(items[0].name);
 
     return (
-        <View borderColor="gray-400" borderWidth="thick" height="90vh" margin="auto" marginTop="size-300" padding="size-100" width="95%">
-            <Calendar events={ events } focusDate={ new Date('March 18 2021') } />
+        <View position="relative">
+            <Flex justifyContent="end" position="absolute" right="0px" top="-2.5rem">
+                <Picker
+                    aria-label={ intl.formatMessage(
+                        { id: 'misc.calendar.label' }) }
+                    items={ items }
+                    onSelectionChange={ (selected) => setCalendarView(selected as string) }
+                    selectedKey={ calendarView }>
+                    { (item) => <Item key={ item.name }>{ item.name }</Item> }
+                </Picker>
+            </Flex>
+            <View height="80vh">
+                { calendarView === items[0].name && 'month calendar' }
+                { calendarView === items[1].name && <Calendar events={ events } focusDate={ new Date('March 18 2021') } /> }
```

Comment:
Because you don't have a compile-time constant for the view state (month or week), you have to resort to comparing against the hard-coded indices of the month and week strings, i.e. 0 and 1 respectively. This is brittle, because if you change the order in your `items` array, these conditions are now flipped and will be returning the month calendar when week is selected, and vice versa.

It doesn't have to be a huge issue because if someone wants you to change the order in the picker (so that week is on top of month for example), you can just make the same change here.

But what if the designer comes back and says they should be ordered alphabetically? Then you need to do a sort post-translation, and depending on language either month or week may end up on top. In those cases, comparing against the index won't work.

That's a good reason to use some sort of hard-coded machine-readable keys like `'month'` and `'week'` instead. You can do it the way I suggested above in a dictionary, or using the array format you already have, but pairing the localized strings with something that remains constant regardless of language.